src/parse/converters/expressions/primary/literal/objectLiteral/keyValuePair.js

```diff
@@ -10,6 +10,8 @@ export default function readKeyValuePair ( parser ) {
 	// allow whitespace between '{' and key
 	parser.allowWhitespace();
 
+	const refKey = parser.nextChar() !== '\'' && parser.nextChar() !== '"';
```

Comment:
It won't if you forget to switch back to the right branch before checking :)

Thanks and fixed..